xml_converter/src/category_gen.cpp

```diff
@@ -111,8 +113,22 @@ waypoint::Category Category::as_protobuf() const {
     if (this->tooltip_description_is_set) {
         proto_category.set_tip_description(this->tooltip_description);
     }
+
+    auto poi = parsed_pois->find(full_category_name);
+
+    if (poi != parsed_pois->end()) {
+        if (poi->second->classname() == "POI") {
+            Icon* icon = dynamic_cast<Icon*>(poi->second);
+            proto_category.add_icon()->MergeFrom(icon->as_protobuf());
```

Comment:
Is this only including a single POI or Trail per category? There can be multiple per category.